apps/omg/lib/omg/fees.ex

```diff
@@ -67,4 +105,119 @@ defmodule OMG.Fees do
+      ...>  },
+      ...>  "omg" => %{
+      ...>    amount: 3,
+      ...>    pegged_amount: 4,
+      ...>    pegged_currency: "USD",
+      ...>    pegged_subunit_to_unit: 100,
+      ...>    updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+      ...>  }
+      ...> },
+      ...> ["eth"]
+      ...> )
+      {:ok,
+        %{
+          "eth" =>
+          %{
+            amount: 1,
+            pegged_amount: 4,
+            pegged_currency: "USD",
+            pegged_subunit_to_unit: 100,
+            updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+          }
+        }
+      }
+
+  """
+  @spec filter_fees(fee_t(), list(String.t())) :: {:ok, fee_t()} | {:error, :currency_fee_not_supported}
+  # empty list = no filter
+  def filter_fees(fees, []), do: {:ok, fees}
+
+  def filter_fees(fees, desired_currencies) do
+    Enum.reduce_while(desired_currencies, {:ok, %{}}, fn currency, {:ok, filtered_fees} ->
+      case Map.fetch(fees, currency) do
+        :error -> {:halt, {:error, :currency_fee_not_supported}}
+        {:ok, fee} -> {:cont, {:ok, Map.put(filtered_fees, currency, fee)}}
+      end
+    end)
+  end
+
+  @doc ~S"""
+  Formats the given fees for an api response.
+
+  ## Examples
+
+      iex> OMG.Fees.to_api_format(
+      ...> %{
+      ...>  "eth" => %{
+      ...>    amount: 1,
+      ...>    pegged_amount: 4,
+      ...>    pegged_currency: "USD",
+      ...>    pegged_subunit_to_unit: 100,
+      ...>    updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+      ...>  },
+      ...>  "omg" => %{
+      ...>    amount: 3,
+      ...>    pegged_amount: 4,
+      ...>    pegged_currency: "USD",
+      ...>    pegged_subunit_to_unit: 100,
+      ...>    updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+      ...>  }
+      ...> })
+      [
+        %{
+          currency: "eth",
+          amount: 1,
+          pegged_amount: 4,
+          pegged_currency: {:skip_hex_encode, "USD"},
+          pegged_subunit_to_unit: 100,
+          updated_at: {:skip_hex_encode, DateTime.from_iso8601("2019-01-01T10:10:00+00:00")}
+        },
+        %{
+          currency: "omg",
+          amount: 3,
+          pegged_amount: 4,
+          pegged_currency: {:skip_hex_encode, "USD"},
+          pegged_subunit_to_unit: 100,
+          updated_at: {:skip_hex_encode, DateTime.from_iso8601("2019-01-01T10:10:00+00:00")}
+        },
+      ]
+
+  """
+  @spec to_api_format(fee_t()) :: list(map())
+  def to_api_format(fees) do
+    Enum.map(fees, fn {currency,
+                       %{
+                         amount: amount,
+                         pegged_currency: pegged_currency,
+                         pegged_amount: pegged_amount,
+                         pegged_subunit_to_unit: pegged_subunit_to_unit,
+                         updated_at: updated_at
+                       }} ->
+      %{
+        currency: currency,
+        amount: amount,
```

Comment:
this might be more readable if you `data.amount`, `data.pegged_currency` ... instead of pattern matching it